src/wallet/api/schemas/walletApi.ts

```diff
@@ -0,0 +1,513 @@
+    authorities: z.object({
+      mint: z.boolean(),
+      melt: z.boolean(),
+    }),
+  }),
+});
+
+/**
+ * Schema for token information.
+ */
+export const tokenInfoSchema = z.object({
+  id: z.string(),
+  name: z.string(),
+  symbol: z.string(),
+});
+
+/**
+ * Schema for token balance information.
+ * Represents both unlocked and locked balances for a token.
+ */
+export const balanceSchema = z.object({
+  unlocked: bigIntCoercibleSchema,
+  locked: bigIntCoercibleSchema,
+});
+
+/**
+ * Schema for token authority balances.
+ * Represents mint and melt authority balances in both unlocked and locked states.
+ */
+export const authorityBalanceSchema = z.object({
+  unlocked: z.object({
+    mint: z.boolean(),
+    melt: z.boolean(),
+  }),
+  locked: z.object({
+    mint: z.boolean(),
+    melt: z.boolean(),
+  }),
+});
+
+/**
+ * Schema for balance object.
+ * Contains token info, balance, authorities, and transaction count.
+ */
+export const getBalanceObjectSchema = z.object({
+  token: tokenInfoSchema,
+  balance: balanceSchema,
+  tokenAuthorities: authorityBalanceSchema,
+  transactions: z.number(),
+  lockExpires: z.number().nullable(),
+});
+
+/**
+ * Response schema for token balances.
+ * Contains an array of balance objects for each token.
+ */
+export const balanceResponseSchema = baseResponseSchema.extend({
+  balances: z.array(getBalanceObjectSchema),
+});
+
+/**
+ * Schema for transaction proposal inputs.
+ * Represents the inputs that will be used in a transaction.
+ */
+export const txProposalInputsSchema = z.object({
+  txId: z.string(),
+  index: z.number(),
+  addressPath: z.string(),
+});
+
+/**
+ * Schema for transaction proposal outputs.
+ * Represents the outputs that will be created in a transaction.
+ */
+export const txProposalOutputsSchema = z.object({
+  address: z.string(),
+  value: bigIntCoercibleSchema,
+  token: z.string(),
+  timelock: z.number().nullable(),
+});
+
+/**
+ * Response schema for creating a transaction proposal.
+ * Contains the proposal ID and the transaction details.
+ */
+export const txProposalCreateResponseSchema = baseResponseSchema.extend({
+  txProposalId: z.string(),
+  inputs: z.array(txProposalInputsSchema),
+});
+
+/**
+ * Response schema for updating a transaction proposal.
+ * Contains the proposal ID and the transaction hex.
+ */
+export const txProposalUpdateResponseSchema = baseResponseSchema.extend({
+  txProposalId: z.string(),
+  txHex: z.string(),
+});
+
+/**
+ * Schema for full node version data.
+ * Contains network parameters and configuration values.
+ */
+export const fullNodeVersionDataSchema = z.object({
+  timestamp: z.number(),
+  version: z.string(),
+  network: z.string(),
+  minWeight: z.number(),
+  minTxWeight: z.number(),
+  minTxWeightCoefficient: z.number(),
+  minTxWeightK: z.number(),
+  tokenDepositPercentage: z.number(),
+  rewardSpendMinBlocks: z.number(),
+  maxNumberInputs: z.number(),
+  maxNumberOutputs: z.number(),
+});
```

Comment:
Currently any new keys will be ignored, but we can pass them into the final data without validation (so that new keys from the fullnode do not break the api)

```suggestion
}).passthrough();
```

Reply:
Done!

Also added a test to test for new attributes